File: app/src/main/java/com/singword/app/data/local/wordbook/WordbookMeta.kt
package com.singword.app.data.local.wordbook

data class WordbookMeta(
    val id: WordbookId,
    val assetName: String,
    val label: String
)

object WordbookCatalog {
    val all = listOf(
        WordbookMeta(WordbookId.CET4, "cet4", "CET-4"),
        WordbookMeta(WordbookId.CET6, "cet6", "CET-6"),
        WordbookMeta(WordbookId.IELTS, "ielts", "IELTS"),
        WordbookMeta(WordbookId.TOEFL, "toefl", "TOEFL")
    )

    fun metaFor(id: WordbookId): WordbookMeta = all.first { it.id == id }
}

File: ios/singword/Data/Wordbooks/WordbookMeta.swift
import Foundation

struct WordbookMeta: Hashable {
    let id: WordbookId
    let assetName: String
    let label: String
}

enum WordbookCatalog {
    nonisolated static let all: [WordbookMeta] = [
        WordbookMeta(id: .cet4, assetName: "cet4", label: "CET-4"),
        WordbookMeta(id: .cet6, assetName: "cet6", label: "CET-6"),
        WordbookMeta(id: .ielts, assetName: "ielts", label: "IELTS"),
        WordbookMeta(id: .toefl, assetName: "toefl", label: "TOEFL")
    ]

    nonisolated static func meta(for id: WordbookId) -> WordbookMeta {
        all.first { $0.id == id } ?? WordbookMeta(id: .cet4, assetName: "cet4", label: "CET-4")
    }
}
